task-sdk/src/airflow/sdk/execution_time/secrets_masker.py

```diff
@@ -345,7 +345,10 @@ def add_mask(self, secret: str | dict | Iterable, name: str | None = None):
                         new_mask = True
 
             if new_mask:
-                self.replacer = re.compile("|".join(self.patterns))
+                # Match pattern followed by either end of string or non-alphanumeric character
+                # This prevents matching substrings within words while handling special characters
+                pattern = r"\b(" + "|".join(self.patterns) + r")(?=$|[^a-zA-Z0-9])"
+                self.replacer = re.compile(pattern)
```

Comment:
I wonder what the thought was here. Do we continue to do partial matches or do it with a word boundary?

Reply:
cc @kaxil @ashb

Reply:
Why was the original implementation using partial matching (where any occurrence of a sensitive value would be masked, even as a substring) instead of word boundary matching?